litmus-portal/graphql-server/pkg/file_handlers/file_handlers.go

```diff
@@ -28,56 +26,65 @@ var subscriberConfiguration = &types.SubscriberConfigurationVars{
 //FileHandler dynamically generates the manifest file and sends it as a response
 func FileHandler(w http.ResponseWriter, r *http.Request) {
 	var (
-		vars           = mux.Vars(r)
-		token          = vars["key"]
+		vars  = mux.Vars(r)
+		token = vars["key"]
+	)
+
+	response, statusCode, err := GetManifest(token)
+	if err != nil {
+		log.Print("error", err)
+		utils.WriteHeaders(&w, statusCode)
+	}
+
+	utils.WriteHeaders(&w, statusCode)
+	w.Write(response)
+}
+
+func GetManifest(token string) ([]byte, int, error) {
+	var (
 		portalEndpoint string
 	)
 
 	id, err := cluster.ClusterValidateJWT(token)
 	if err != nil {
-		log.Print("ERROR", err)
-		utils.WriteHeaders(&w, 404)
-		return
+		return nil, 404, err
 	}
 
 	reqCluster, err := database.GetCluster(id)
 	if err != nil {
-		log.Print("ERROR", err)
-		utils.WriteHeaders(&w, 500)
-		return
+		return nil, 500, err
 	}
 
 	if os.Getenv("PORTAL_SCOPE") == "cluster" {
 		portalEndpoint, err = k8s.GetPortalEndpoint()
 		if err != nil {
-			log.Print(err)
+			return nil, 500, err
 		}
 	} else if os.Getenv("PORTAL_SCOPE") == "namespace" {
 		portalEndpoint = os.Getenv("PORTAL_ENDPOINT")
 	}
 
 	subscriberConfiguration.GQLServerURI = portalEndpoint + "/query"
-
 	if !reqCluster.IsRegistered {
 		var respData []byte
 
 		if reqCluster.AgentScope == "cluster" {
 			respData, err = utils.ManifestParser(reqCluster, "manifests/cluster-subscriber.yml", subscriberConfiguration)
+			if err != nil {
+				return nil, 500, err
+			}
 		} else if reqCluster.AgentScope == "namespace" {
 			respData, err = utils.ManifestParser(reqCluster, "manifests/namespace-subscriber.yml", subscriberConfiguration)
+			if err != nil {
```

Comment:
move the error checkout out of this if, no need to repeat.

Reply:
Here, err is local scope inside the if condition. If we need to move `err` check condition outside of if block, then we have to make `err` variable globally.

Reply:
No err is already declared in the global scope line 48

Reply:
But this can be done by making in global inside that function. Not a big issue

Reply:
The line doesn't use `:=` rather uses `=`

Reply:
we need to use = to update the global err